src/zmalloc.c

```diff
@@ -124,33 +346,36 @@ void *zmalloc_no_tcache(size_t size) {
     ASSERT_NO_SIZE_OVERFLOW(size);
     void *ptr = mallocx(size+PREFIX_SIZE, MALLOCX_TCACHE_NONE);
     if (!ptr) zmalloc_oom_handler(size);
-    update_zmalloc_stat_alloc(zmalloc_size(ptr));
+    update_zmalloc_dram_stat_alloc(zmalloc_size(ptr));
     return ptr;
 }
 
 void zfree_no_tcache(void *ptr) {
     if (ptr == NULL) return;
-    update_zmalloc_stat_free(zmalloc_size(ptr));
+    update_zmalloc_dram_stat_free(zmalloc_size(ptr));
     dallocx(ptr, MALLOCX_TCACHE_NONE);
 }
 #endif
 
-void *zcalloc(size_t size) {
-    ASSERT_NO_SIZE_OVERFLOW(size);
```

Comment:
Don't remove this assert. In fact, I think you can safely add this assert to zmalloc_pmem(), zcalloc_pmem() and zrealloc_pmem() as well.

Reply:
Done